core/components/atoms/dropdown/DropdownList.tsx

```diff
@@ -463,17 +465,21 @@ const DropdownList = (props: OptionsProps) => {
     const { selectAllLabel = 'Select All', selectAll, onSelectAll } = props;
 
     const label = selectAllLabel.trim() ? selectAllLabel.trim() : 'Select All';
+    const [checkboxId] = React.useState(() => uidGenerator());
 
     return (
       <div className={SelectAllClass} onMouseEnter={() => updateActiveOption(0, true)}>
-        <Checkbox
-          label={label}
-          onChange={onSelectAll}
-          checked={selectAll.checked}
-          indeterminate={selectAll.indeterminate}
-          tabIndex={-1}
-          className="OptionCheckbox"
-        />
+        <label htmlFor={checkboxId} className="Checkbox-label">
+          <Checkbox
```

Comment:
The checkbox component inside the label does not seem to be semantically correct. please follow this [thread](https://stackoverflow.com/questions/774054/should-i-put-input-elements-inside-a-label-element). 

or If you include the checkbox tag in the label tag, you don't need to use the 'for' attribute.